src/runtime/crt/common/func_registry.c

```diff
@@ -60,14 +60,20 @@ int strcmp_cursor(const char** cursor, const char* name) {
   return return_value;
 }
 
+uint16_t* TVMFuncRegistry_GetNumFuncs(const TVMFuncRegistry* reg) { return (uint16_t*)reg->names; }
```

Comment:
for this function, i think we need to impl like this to avoid alignment problems:
```suggestion
uint16_t TVMFuncRegistry_GetNumFuncs(const TVMFuncRegistry* reg) { 
  uint16_t num_funcs;
  memcpy(&num_funcs, reg->names, sizeof(num_funcs));
  return num_funcs;
  }
```

Reply:
ah, yeah i think so. good catch.

Reply:
@areusch I have implemented the functions `TVMFuncRegistry_SetNumFuncs` and `TVMFuncRegistry_GetNumFuncs`.
You can review these functions.